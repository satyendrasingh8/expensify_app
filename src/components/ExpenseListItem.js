import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

 const ExpenseListItem = ({id,description,amount,createdAt}) =>(
    <div>
    <Link to={`/edit/${id}`}>
<h2>{description} </h2>
   </Link>
  <p>
  {numeral(amount/100).format('$0,0.00')} 
    <strong> -  </ strong>
   {moment(createdAt).format('ddd MMM Do,YYYY')}  
   </p>

    </div>
);

export default ExpenseListItem;