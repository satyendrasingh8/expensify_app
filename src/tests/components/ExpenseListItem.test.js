import React from 'react';
import {shallow} from 'enzyme';
import expenses from  '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem'


test('should render expenseListItem from expenses', () => {
const wrapper = shallow(<ExpenseListItem expenses={expenses}  />)
expect(wrapper).toMatchSnapshot();
})