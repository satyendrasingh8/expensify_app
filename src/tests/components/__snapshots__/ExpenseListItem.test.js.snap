// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseListItem from expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseListItem
    expenses={
        Array [
            Object {
              "amount": 1500,
              "createdAt": 0,
              "description": "hello",
              "id": "1",
              "note": "",
            },
            Object {
              "amount": 1300,
              "createdAt": -950400000,
              "description": "hi",
              "id": "2",
              "note": "",
            },
            Object {
              "amount": 400,
              "createdAt": 864000000,
              "description": "hope",
              "id": "3",
              "note": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          to="/edit/undefined"
>
          <h2>
                     
          </h2>
</Unknown>,
        <p>
          $0.00
          <strong>
                     -  
          </strong>
          Thu Jan 1st,1970
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h2>
             
</h2>,
          "to": "/edit/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
          ],
          "type": "h2",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "$0.00",
            <strong>
               -  
</strong>,
            "Thu Jan 1st,1970",
          ],
        },
        "ref": null,
        "rendered": Array [
          "$0.00",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " -  ",
            },
            "ref": null,
            "rendered": " -  ",
            "type": "strong",
          },
          "Thu Jan 1st,1970",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            to="/edit/undefined"
>
            <h2>
                         
            </h2>
</Unknown>,
          <p>
            $0.00
            <strong>
                         -  
            </strong>
            Thu Jan 1st,1970
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h2>
               
</h2>,
            "to": "/edit/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
            ],
            "type": "h2",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "$0.00",
              <strong>
                 -  
</strong>,
              "Thu Jan 1st,1970",
            ],
          },
          "ref": null,
          "rendered": Array [
            "$0.00",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " -  ",
              },
              "ref": null,
              "rendered": " -  ",
              "type": "strong",
            },
            "Thu Jan 1st,1970",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
